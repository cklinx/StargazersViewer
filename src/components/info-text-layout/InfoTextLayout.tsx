import React from 'react';
import {StyleSheet, Text, View, ViewProps} from 'react-native';
import {styles} from './info-text-layout.presets';
import useColorScheme from '@stargazers/hooks/useColorScheme';
import {Colors} from '@stargazers/theme';
import Icon from 'react-native-vector-icons/Feather';

export const InfoTextLayout: React.FC<
  {
    text: string;
  } & Pick<ViewProps, 'style'>
> = ({text, style}) => {
  const colorScheme = useColorScheme();
  return (
    <View
      key={text}
      style={StyleSheet.flatten([
        styles.infoRoot,
        {backgroundColor: Colors[colorScheme].infoTextBackground},
        style,
      ])}>
      <Icon name={'alert-circle'} size={18} color={Colors[colorScheme].infoTextColor} />
      <Text
        style={{
          ...styles.infoText,
          color: Colors[colorScheme].infoTextColor,
        }}>
        {text}
      </Text>
    </View>
  );
};
