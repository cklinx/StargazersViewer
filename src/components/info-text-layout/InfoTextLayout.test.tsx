import React from 'react';

import renderer from 'react-test-renderer';
import {InfoTextLayout} from './InfoTextLayout';

jest.mock('react-native-vector-icons/Feather', () => 'Feather');
describe('InfoText', () => {
  it('should render correctly', () => {
    const {toJSON} = renderer.create(<InfoTextLayout text="Text" />);

    expect(toJSON()).toMatchSnapshot();
  });
});
