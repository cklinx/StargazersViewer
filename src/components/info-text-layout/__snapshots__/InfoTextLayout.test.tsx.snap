// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoText should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#9867F0",
      "borderRadius": 12,
      "flexDirection": "row",
      "marginTop": 12,
      "padding": 16,
      "width": "100%",
    }
  }
>
  <Feather
    color="#FFFFFF"
    name="alert-circle"
    size={18}
  />
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "HelveticaNeue-Medium",
        "fontSize": 16,
        "marginEnd": 16,
        "marginStart": 8,
      }
    }
  >
    Text
  </Text>
</View>
`;
